### PR TITLE
Create client -> project -> task hierarchy

### PR DESCRIPTION
**Pull request by hargup**
dependent on https://github.com/nilenso/time-tracker/pull/172

### CODE REVIEW

File: src/cljs/time_tracker_web_nxt/config.cljs [hunk 1]
```diff
@@ -8,9 +8,9 @@
 (goog-define ^boolean debug? false)
 
 (def dev
-  {:client-id "163789226845-gfp9na27iac63lnrb81hsdkvf5log85i.apps.googleusercontent.com"
```
Comment (samrat) [inline src/cljs/time_tracker_web_nxt/config.cljs]: This should not be committed.

**Comment by kirang89** [comment]
@hargup  @samrat  Please create a card to improve the UI. The changes would be:
- Add some consistent spacing b/w the dropdowns.
- Have a single text field for providing time, like you do in Harvest.

screen shot 2018-04-25 at 11 18 49 am

File: src/cljs/time_tracker_web_nxt/views.cljs [hunk 2]
```diff
@@ -17,35 +17,40 @@
...
-  (let [selected-id (:id @selected)]
-    [:select.project-dropdown
-     {:value (or selected-id "")
-      :on-change #(reset! selected {:id (-> % .-target .-value)})}
-     (for [{:keys [id name]} projects]
-       ^{:key id} [:option (if (= selected-id id)
-                             {:value id :label name}
-                             {:value id :label name}) name])]))
+(defn dropdown-widget [items selected-id event]
+  [:select.project-dropdown
```
Comment (kirang89) [inline src/cljs/time_tracker_web_nxt/views.cljs]: If you're creating a generic dropdown widget, do apply a generic class to the element.

File: src/cljs/time_tracker_web_nxt/views.cljs [hunk 3]
```diff
@@ -17,35 +17,40 @@
...
-        data            (reagent/atom {:notes ""
+  (let [data            (reagent/atom {:notes ""
                                        :elapsed-hh 0
                                        :elapsed-mm 0
-                                       :elapsed-ss 0})
-        show?            (rf/subscribe [:show-create-timer-widget?])]
+                                       :elapsed-ss 0})]
     (fn []
-      (let [default-selected     {:id (:id (first @projects))}
+      (let [show?            (rf/subscribe [:show-create-timer-widget?])
```
Comment (kirang89) [inline src/cljs/time_tracker_web_nxt/views.cljs]: In general, it's a good practice to align the bindings inside a let for the sake of readability.

**Comment by hargup** [comment]
@kirang89 Thanks for the comments, have created an issue for UI changes here https://github.com/nilenso/time-tracker/issues/179

File: src/cljs/time_tracker_web_nxt/views.cljs (hunk 3 above)
Comment (hargup) [reply src/cljs/time_tracker_web_nxt/views.cljs]: @kirang89 fixed all the alignment issues in https://github.com/nilenso/time-tracker-web-nxt/pull/44

**Pull request merged**